File: packages/tools/src/shared/forget-memory.ts
const DEFAULT_BASE_URL = "https://api.supermemory.ai"
const FETCH_TIMEOUT_MS = 30_000

export interface ForgetMemoryParams {
	containerTag: string
	id?: string
	content?: string
	reason?: string
}

export interface ForgetMemoryRequestOptions {
	signal?: AbortSignal
}

/**
 * Marks a memory as forgotten via `DELETE /v4/memories`.
 *
 * The supermemory SDK version this package depends on (v3) has no
 * `memories.forget` method, so the endpoint is called directly — the same
 * pattern the middleware already uses for `/v4/profile` and
 * `/v4/conversations`.
 */
export async function forgetMemoryRequest(
	apiKey: string,
	params: ForgetMemoryParams,
	baseUrl: string = DEFAULT_BASE_URL,
	options?: ForgetMemoryRequestOptions,
): Promise<void> {
	const response = await fetch(`${baseUrl}/v4/memories`, {
		method: "DELETE",
		headers: {
			"Content-Type": "application/json",
			Authorization: `Bearer ${apiKey}`,
		},
		body: JSON.stringify(params),
		signal: options?.signal ?? AbortSignal.timeout(FETCH_TIMEOUT_MS),
	})

	if (!response.ok) {
		const errorText = await response.text().catch(() => "Unknown error")
		throw new Error(
			`Supermemory forget memory failed: ${response.status} ${response.statusText}. ${errorText}`,
		)
	}
}

File: packages/openai-sdk-python/src/supermemory_openai/forget_memory.py
"""Forget memory via DELETE /v4/memories (not exposed on supermemory SDK v3)."""

from typing import Optional

DEFAULT_BASE_URL = "https://api.supermemory.ai"


async def forget_memory_request(
    api_key: str,
    container_tag: str,
    memory_id: Optional[str] = None,
    memory_content: Optional[str] = None,
    reason: Optional[str] = None,
    base_url: str = DEFAULT_BASE_URL,
) -> None:
    """Mark a memory as forgotten via the v4 memories endpoint."""
    payload: dict[str, str] = {"containerTag": container_tag}
    if memory_id:
        payload["id"] = memory_id
    if memory_content:
        payload["content"] = memory_content
    if reason:
        payload["reason"] = reason

    try:
        import aiohttp

        async with aiohttp.ClientSession() as session:
            async with session.delete(
                f"{base_url}/v4/memories",
                headers={
                    "Content-Type": "application/json",
                    "Authorization": f"Bearer {api_key}",
                },
                json=payload,
            ) as response:
                if not response.ok:
                    error_text = await response.text()
                    raise RuntimeError(
                        f"Supermemory forget memory failed: {response.status} "
                        f"{response.reason}. {error_text}"
                    )
    except ImportError:
        import requests

        response = requests.delete(
            f"{base_url}/v4/memories",
            headers={
                "Content-Type": "application/json",
                "Authorization": f"Bearer {api_key}",
            },
            json=payload,
            timeout=30,
        )
        if not response.ok:
            raise RuntimeError(
                f"Supermemory forget memory failed: {response.status_code} "
                f"{response.reason}. {response.text}"
            )
